docs/ai/llm-spans.md

```diff
@@ -0,0 +1,99 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: LLM Calls
+--->
+
+# Semantic Conventions for LLM requests
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [LLM Request attributes](#llm-request-attributes)
+- [Configuration](#configuration)
+- [Semantic Conventions for specific LLM technologies](#semantic-conventions-for-specific-llm-technologies)
+
+<!-- tocstop -->
+
+A request to an LLM is modeled as a span in a trace.
+
+The **span name** SHOULD be set to a low cardinality value representing the request made to an LLM.
+It MAY be a name of the API endpoint for the LLM being called.
+
+## Configuration
+
+Instrumentations for LLMs MUST offer the ability to turn off capture of prompts and completions. This is for three primary reasons:
+
+1. Data privacy concerns. End users of LLM applications may input sensitive information or personally identifiable information (PII) that they do not wish to be sent to a telemetry backend.
+2. Data size concerns. Although there is no specified limit to sizes, there are practical limitations in programming languages and telemety systems. Some LLMs allow for extremely large context windows that end users may take full advantage of.
+3. Performance concerns. Sending large amounts of data to a telemetry backend may cause performance issues for the application.
+
+By default, these configurations SHOULD NOT capture prompts and completions.
+
+## LLM Request attributes
+
+These attributes track input data and metadata for a request to an LLM. Each attribute represents a concept that is common to most LLMs.
+
+<!-- semconv ai(tag=llm-request) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `llm.vendor` | string | The name of the LLM foundation model vendor, if applicable. If not using a vendor-supplied model, this field is left blank. | `openai` | Recommended |
+| `llm.request.model` | string | The name of the LLM a request is being made to. If the LLM is supplied by a vendor, then the value must be the exact name of the model requested. If the LLM is a fine-tuned custom model, the value SHOULD have a more specific name than the base model that's been fine-tuned. | `gpt-4` | Required |
+| `llm.request.max_tokens` | int | The maximum number of tokens the LLM generates for a request. | `100` | Recommended |
```

Comment:
`max_tokens` is prefixed with `request`, whereas other parameters such as temperature are not prefixed. Perhaps we should remove the `request` prefix or add the prefix to the others. Instead of `request`, perhaps `parameter` is better.

Reply:
Hmm... Parameter would sound weird for `model`, no? `llm.parameter.model`. I've added `request` to all `request` parameters.